WalletWasabi/Blockchain/TransactionBuilding/SmartCoinSelector.cs

```diff
@@ -65,21 +65,21 @@ public IEnumerable<ICoin> Select(IEnumerable<ICoin> coins, IMoney target)
 
 		var coinsInBestClusterByScript = bestCoinCluster
 			.GroupBy(c => c.ScriptPubKey)
-			.Select(group => new { Coins = group });
+			.Select(group => (ScriptPubKey: group.Key, Coins: group.ToList()))
+			.OrderBy(x => x.Coins.Sum(c => c.Amount))
+			.ToImmutableList();
 
-		var coinsToSpend = new List<SmartCoin>();
+		var coinsGroup = coinsInBestClusterByScript.Count < 10
+			? coinsInBestClusterByScript.CombinationsWithoutRepetition(1, 5).Select(x => x.ToImmutableList())
+			: coinsInBestClusterByScript.Select(x => ImmutableList.Create(x))
```

Comment:
I am not sure what is happening here. ImmutableList syntax worsens (for me) the readability. Can you add a few comments on what is happening here?